File: serialization/java/json/lib/src/main/java/com/microsoft/kiota/serialization/JsonParseNodeFactory.java
package com.microsoft.kiota.serialization;

import com.microsoft.kiota.serialization.ParseNode;
import com.microsoft.kiota.serialization.ParseNodeFactory;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.nio.charset.StandardCharsets;
import java.util.Objects;
import java.util.stream.Collectors;

import javax.annotation.Nonnull;

public class JsonParseNodeFactory implements ParseNodeFactory {
    public String getValidContentType() {
        return validContentType;
    }
    private final static String validContentType = "application/json";
    @Override
    @Nonnull
    public ParseNode getParseNode(@Nonnull final String contentType, @Nonnull final InputStream rawResponse) {
        Objects.requireNonNull(contentType, "parameter contentType cannot be null");
        Objects.requireNonNull(rawResponse, "parameter rawResponse cannot be null");
        if(contentType.isEmpty()) {
            throw new NullPointerException("contentType cannot be empty");
        } else if (!contentType.equals(validContentType)) {
            throw new IllegalArgumentException("expected a " + validContentType + " content type");
        }
        String rawText;
        try(final InputStreamReader reader = new InputStreamReader(rawResponse, StandardCharsets.UTF_8)) {
            try(final BufferedReader buff = new BufferedReader(reader)) {
                rawText = buff.lines()
                .collect(Collectors.joining("\n"));
            }
        } catch (IOException ex) {
            throw new RuntimeException("could not close the reader", ex);
        }
        return new JsonParseNode(rawText);
    }

}

File: serialization/php/json/src/JsonParseNodeFactory.php
<?php

namespace Microsoft\Kiota\Serialization\Json;

use JsonException;
use InvalidArgumentException;
use Microsoft\Kiota\Abstractions\Serialization\ParseNode;
use Microsoft\Kiota\Abstractions\Serialization\ParseNodeFactory;
use Psr\Http\Message\StreamInterface;
use RuntimeException;

class JsonParseNodeFactory implements ParseNodeFactory
{

    /**
     * @inheritDoc
     */
    public function getRootParseNode(string $contentType, StreamInterface $rawResponse): ParseNode {
        if (empty($contentType)) {
            throw new InvalidArgumentException('$contentType cannot be empty.');
        }

        $streamContents = $rawResponse->getContents();
        if (strcasecmp($this->getValidContentType(), $contentType) !== 0){
            throw new InvalidArgumentException("expected a {$this->getValidContentType()} content type.");
        }
        if (empty($streamContents)){
            throw new InvalidArgumentException('$rawResponse cannot be empty.');
        }
        try {
            $content = json_decode($streamContents, true, 512, JSON_THROW_ON_ERROR);
        } catch (JsonException $ex){
            throw new RuntimeException('The was a problem parsing the response.', 1, $ex);
        }
        return new JsonParseNode($content);
    }

    /**
     * @inheritDoc
     */
    public function getValidContentType(): string {
        return 'application/json';
    }
}